src/components/markdown/markdown.tsx

```diff
@@ -48,25 +49,42 @@ export class Markdown {
     public whitelist?: CustomElementDefinition[] =
         globalConfig.markdownWhitelist;
 
+    /**
+     * Enable lazy loading for images
+     */
+    @Prop()
+    public lazyLoadImages = false;
+
     @Watch('value')
     public async textChanged() {
         try {
+            this.cleanupImageIntersectionObserver();
+
             const html = await markdownToHTML(this.value, {
                 forceHardLineBreaks: true,
                 whitelist: this.whitelist ?? [],
+                lazyLoadImages: this.lazyLoadImages,
             });
+
             this.rootElement.innerHTML = html;
+
+            this.setupImageIntersectionObserver();
         } catch (error) {
             // eslint-disable-next-line no-console
             console.error(error);
         }
     }
 
+    private rootElement: HTMLDivElement;
+    private imageIntersectionObserver: ImageIntersectionObserver;
```

Comment:
Let's fix the type since it's set to `null` in the cleanup method :slightly_smiling_face: 
 
```suggestion
    private imageIntersectionObserver: ImageIntersectionObserver | null = null;
```